lib/ansible/modules/network/edgeos/edgeos_config.py

```diff
@@ -206,67 +242,118 @@ def check_command(module, command):
+                break
+
+    unmanaged_config = [line for line in unmanaged_config if line not in matches]
+
+    return updates, unmanaged_config, invalid_commands
 
-            if item not in config:
-                updates.append(line)
 
-            # If there is a corresponding delete command in the desired config, make sure to append
-            # the set command even though it already exists in the running config
-            else:
-                ditem = re.sub('set', 'delete', item)
-                for line in delete_commands:
-                    if ditem.startswith(line):
-                        updates.append(item)
-
-        elif item.startswith('delete'):
-            if not config:
-                updates.append(line)
-            else:
-                item = re.sub(r'delete', 'set', item)
-                for entry in config:
-                    if entry.startswith(item) and line not in visited:
-                        updates.append(line)
-                        visited.add(line)
+def run(module, result):
+    """compares config against passed configuration to asnible to create an
+    update list and applies to the edgeos device.
 
-    return list(updates)
+    .. warning:: docstring added long after code written, requires verification
+    of Arguments and Returns - please update if you see any errors
 
+    :param module: ansible module for self ref
+    :type module: ansible.module
+    :param result: result dict to be populated
+    process
+    :type result: dict
+    """
 
-def run(module, result):
     # get the current active config from the node or passed in via
     # the config param
     config = module.params['config'] or get_config(module)
 
     # create the candidate config object from the arguments
     candidate = get_candidate(module)
 
-    # create loadable config that includes only the configuration updates
-    commands = diff_config(module, candidate, config)
+    # create loadable config from updates, also return unmanaged and invalid
+    # commands
+    updates, unmanaged_config, invalid_commands = diff_config(module, candidate, config)
 
-    result['commands'] = commands
+    result['commands'] = sorted(updates)
+    result['unmanaged'] = sorted(unmanaged_config)
+    result['invalid'] = sorted(invalid_commands)
 
     commit = not module.check_mode
     comment = module.params['comment']
 
-    if commands:
-        load_config(module, commands, commit=commit, comment=comment)
-
+    if result.get('commands'):
+        load_config(module, updates, commit=commit, comment=comment)
         result['changed'] = True
 
+    if result.get('unmanaged'):
+        result['warnings'].append('Some configuration commands were '
+                                  'unmanaged, review unmanaged list')
+
+    if result.get('invalid'):
+        result['warnings'].append('Some configuration commands were '
+                                  'invalid, review invalid list')
```

Comment:
There's no need to wrap the strings like this. Our project lint settings accept up to 160 characters wide.
```suggestion
        result['warnings'].append('Some configuration commands were unmanaged, review unmanaged list')

    if result.get('invalid'):
        result['warnings'].append('Some configuration commands were invalid, review invalid list')
```